datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -515,14 +517,29 @@ fn like_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
 /// casted to for the purpose of a date computation
 fn temporal_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType> {
     use arrow::datatypes::DataType::*;
-    use arrow::datatypes::TimeUnit;
     match (lhs_type, rhs_type) {
         (Date64, Date32) => Some(Date64),
         (Date32, Date64) => Some(Date64),
         (Utf8, Date32) => Some(Date32),
         (Date32, Utf8) => Some(Date32),
         (Utf8, Date64) => Some(Date64),
         (Date64, Utf8) => Some(Date64),
+        (Utf8, Time32(unit)) => match check_time_unit(Time32(unit.clone())) {
```

Comment:
**nitpick:** You can avoid a clone here by passing a reference

Reply:
Hummm, unless there is something I am not aware of, I don't see how to get rid of these cloning operations. The problem is that the `match` already takes `unit` as `&TimeUnit`, and then I have to define `Time32` with a `TimeUnit`, hence I need to take ownership of the content `unit` points to. I tried to pass it as a reference nevertheless (and use it as a reference within the function), but it doesn't work due to the definition of the `Time32` type in `datatypes.rs`.

Reply:
Ah I see what you mean. It's fine then